keps/sig-network/3015-prefer-same-node/README.md

```diff
@@ -0,0 +1,390 @@
+and not receiving unexpected connections.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+The implementation of the feature itself has no SLOs. The effect it
+has on the performance of end user workloads that use the feature
+depends on those workloads.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+The implementation of the feature itself has no SLIs, other than the
+generic kube-proxy metrics. User workloads that use the feature may
+expose SLI information that the user can examine to determine how well
+the feature is working for their workload.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+Not really; we don't know how fast the user's services are supposed to
+be, so we can't really tell if we are improving them as much as they
+hoped or not.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+It depends on a service proxy which recognizes the new traffic policy
+values. We will update `kube-proxy` ourselves, but network plugins /
+kubernetes distributions that ship their own alternative service
+proxies will also need to be updated to support the new value before
+their users can make use of it. (Until then, `TrafficDistribution:
+PreferSameNode` would be implemented as `TrafficDistribution:
+PreferClose`.)
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No (other than that it means people may set `TrafficDistribution` on
+Services where they were not previously setting it).
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+No change from existing service/proxy behavior.
+
+###### What are other known failure modes?
+
+None known
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+N/A
+
+## Implementation History
+
+- Initial proposal as `InternalTrafficPolicy: PreferLocal`: 2021-10-21
+- Initial proposal as "Node-level topology": 2022-01-15
+- Initial proposal as `TrafficDistribution: PreferSameNode`: 2025-02-06
+
+## Drawbacks
+
+## Alternatives
+
+As noted, this is the third attempt at this feature.
+
+The initial proposal ([#3016]) was for `internalTrafficPolicy:
```

Comment:
Don't forget `topologyKeys` which allowed users to express a strict or prefer form of this, but also allowed arbitrary silliness.